myhpom/tasks.py

```diff
@@ -0,0 +1,81 @@
+import json
+import requests
+from django.core.mail import send_mail
+from django.conf import settings
+from django.template.loader import get_template
+from django.template import Context
+from celery import shared_task
+from celery.task import Task
+from celery.signals import task_failure
+from myhpom.models.document import DocumentUrl
+from myhpom.models import cloudfactory
+
+
+@shared_task
+class CloudFactorySubmitDocumentRun(Task):
+    def run(self, document_url_id, document_host=None):
+        """
+        ## Arguments
+        * document_url_id = the database id for the DocumentUrl that is being submitted
+        * document_host = the host (incl. protocol/scheme) to use for the DocumentURL & callback
+            -- configurable so that we can tell CloudFactory which server to interact with
+            (the document_host and the callback_host are assumed the same)
+
+        ## Task Process
+        - submit the DocumentUrl to the CloudFactory (test) endpoint
+        - store the status of the submitted document
+        - if the DocumentUrl no longer exists, abort the task without an Exception.
+            - the DocumentUrl might no longer exist by the time celery picks it up,
+            - or the document it points to might have been removed.
+        - if an error occurs in this process, send support email.
+        """
+        cf_run = cloudfactory.CloudFactoryDocumentRun(document_host=document_host or '')
+
+        # If the DocumentUrl has been deleted, abort the task with a message
+        # -- no need to admin support email, this is an expected case.
+        try:
+            cf_run.document_url = DocumentUrl.objects.get(id=document_url_id)
+        except:
```

Comment:
A few flake8 warnings in here about whitespace and about this bare exception - make this explicitly catch the `DocumentUrl.DoesNotExist` exception that we expect.